source/developer/mobile-developer-setup.rst

```diff
@@ -0,0 +1,143 @@
+
+The following instructions apply to new unreleased mobile apps for iOS and Android built in React Native. They are intended to release in Beta at the end of March 2017. Source code can be found at: https://github.com/mattermost/mattermost-mobile
+
+.. contents:: Contents:
+  :backlinks: top
+  :local:
+
+If you run into any issues getting your environment set up, check the Troubleshooting section at the bottom for common solutions.
+
+Development Environment Setup
+---------------------------------
+
+Mac OS X
+~~~~~~~~~~~~
+
+1. Install `XCode 8.1 <https://developer.apple.com/download/>`_.
+
+2. Install `Homebrew <http://brew.sh/>`_.
+
+3. Using Homebrew, install `Node.js <https://nodejs.org>`_ and npm.
+
+  ``brew install node``
+
+4. Using Homebrew, install `Watchman <https://github.com/facebook/watchman>`_.
+
+  ``brew install watchman``
+
+5. Using npm, install the React Native CLI tools globally.
+
+  ``npm install -g react-native-cli``
+
+6. Fork `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`_ on GitHub.
+
+7. Clone your fork locally.
+
+  ``cd`` into the folder that you want to store the local copy of your code
+
+  ``git clone https://github.com/<username>/mattermost-mobile.git``
+
+  ``cd mattermost-mobile``
+
+8. Using npm, download any other dependencies.
+
+  ``make pre-run``
+
+9. *Optional:* Install Mattermost locally so that you can run unit tests and connect to the server while doing development.
+
+  a. Follow the steps in the `Developer Machine Setup <developer-setup.html>`_ to install Mattermost.
+
+  b. Edit your Mattermost instance's configuration file to allow sign-up without an invite.
+
+    In ``config/config.json``, set ``"EnableOpenServer"`` to ``true``
+
+  c. Start/restart your server.
+
+    ``make restart-server``
+
+Test Environment Setup
+--------------------------
+
+Android (Device)
+~~~~~~~~~~~~~~~~~~~
+
+1. Install the Android SDK (can be skipped if you already have Android Studio installed).
+
+  a. Go to `the Android developer downloads page <https://developer.android.com/studio/index.html#downloads>`_, scroll down to the Get Just the Command Line Tools, and download the zip file suitable for your operating system.
+
+  b. Unzip the SDK to somewhere on your hard drive. For example, ``/Users/<username>/Library/Android/sdk`` on Mac OS X.
+
+  c. Configure the following environment variables:
+
+    - Set ``ANDROID_HOME`` to the location that you unzipped the SDK to.
+
+    - Add ``ANDROID_HOME/tools`` and ``ANDROID_HOME/platform-tools`` to the ``PATH``.
+
+2. Run ``android`` to open the Android SDK Manager and install the following packages:
+
+  - Tools > Android SDK Tools 25.2.5 or higher
+
+  - Tools > Android SDK Platform-tools 25.0.3
+
+  - Tools > Android SDK Build-tools 23.0.1
+
+  - Android 6.0 > SDK Platform 23
+
+  - Android 6.0 > Google APIs 23
+
+  - Android 5.1.1 > SDK Platform 22
+
+  - Android 5.1.1 > Google APIs 22
+
+  - Extras > Android Support Repository
```

Comment:
Same with these libraries. I don't have Google Play Services or the Google Repository